test/test_vector.py

```diff
@@ -834,3 +834,39 @@ def test_numpy_binary():
     assert np.allclose(result.y.values, exp_y)
     assert np.allclose(result.z.values, exp_z)
     assert result.unit == units('m')
+
+
+def test_coordinate_transforms():
+    dataset = Dataset()
+    dataset["amr"] = Datagroup()
+    dataset["test"] = Datagroup()
+    pos_x = [1., 2., 3., 4., 5.]
+    pos_y = [6., 7., 8., 9., 10.]
+    pos_z = [11., 12., 13., 14., 15.]
+    vel_x = 2 * np.array(pos_x)
+    vel_y = 2 * np.array(pos_y)
+    vel_z = 2 * np.array(pos_z)
+    x = Array(values=pos_x, unit='m')
+    y = Array(values=pos_y, unit='m')
+    z = Array(values=pos_z, unit='m')
+    v = Vector(x, y, z)
+    dataset["amr"]["position"] = v
+    dataset["test"]["velocity"] = 2 * v * (1 * units("s^-1"))
+    exp_r = np.linalg.norm([pos_x, pos_y, pos_z], axis=0)
+    exp_theta = np.arctan2(np.linalg.norm([pos_x, pos_y], axis=0), pos_z)
+    exp_phi = np.arctan2(pos_y, pos_x)
+    exp_cyl_r = np.linalg.norm([pos_x, pos_y], axis=0)
+    exp_vr = np.cos(exp_phi) * np.sin(exp_theta) * vel_x + np.sin(exp_phi) * np.sin(
+        exp_theta) * vel_y + np.cos(exp_theta) * vel_z
+    exp_vtheta = np.cos(exp_theta) * np.cos(exp_phi) * vel_x + np.cos(
+        exp_theta) * np.sin(exp_phi) * vel_y - np.sin(exp_theta) * vel_z
+    exp_vphi = -np.sin(exp_phi) * vel_x + np.cos(exp_phi) * vel_y
+    exp_vcyl_r = np.cos(exp_phi) * vel_x + np.sin(exp_phi) * vel_y
```

Comment:
It seems that you just copied an pasted the expressions you have in the Vector class.
While this test will ensure that these results won't change in the future, it doesn't necessarily make sure it is working properly now. I would have used some easy values that we can predict the vector components by hand and check the results are what we expect.

Reply:
How about keeping this test as is and testing that it's working right now in this thread? Since osyris is using a rotation matrix, there's going to be very slight deviations from the analytical formulas (courtesy of the trigonometric functions). What we can do is compare the results of the rotation matrix (`exp_vr`, `exp_vphi`, `exp_vtheta`) with the analytical formulas:

For the radial velocity:

```
In [27]: vr = (vel_x * pos_x + vel_y*pos_y + vel_z * pos_z)/np.linalg.norm([pos_x, pos_y, pos_z], axis=0)                                              
Out[27]: array([25.13961018, 28.0713377 , 31.11269837, 34.23448554, 37.41657387])
In [29]: exp_vr                                                                                                                                   
Out[29]: array([25.13961018, 28.0713377 , 31.11269837, 34.23448554, 37.41657387])
```

For the azimuthal velocity:

```
In [35]: r = np.linalg.norm([pos_x, pos_y, pos_z], axis=0)                                                                                        
                                                                                                                                                  
In [36]: r_cyl = np.linalg.norm([pos_x, pos_y], axis=0)                                                                                           
                                                                                                                                                  
In [37]: vphi = r/r_cyl * (vel_y*pos_x - vel_x*pos_y) * np.sin(np.arccos(pos_z/r))

In [39]: print(vphi, "\n", exp_vphi)                                                                                                              
[0. 0. 0. 0. 0.]                                                                                                                                  
 [-8.88178420e-16 -1.33226763e-15  0.00000000e+00 -8.88178420e-16
  1.77635684e-15]
```

And finally, the meridonial velocity:

```
In [43]: vtheta = (pos_z*vr - vel_z*r)/(r*np.sqrt(1+ pos_z**2 / r**2))
In [45]: print(vtheta, "\n", exp_vtheta)                                                                                                          
[-3.40312647e-15  3.07824371e-15  2.80387908e-15  0.00000000e+00                                                                                  
 -4.74106887e-15] 
 [1.77635684e-15 0.00000000e+00 1.77635684e-15 0.00000000e+00
 0.00000000e+00]
```

You can see that the rotation matrix has deviations from the analytical formulas of the order of 10^-15 ~10^-16. Although this is acceptable, I think it might cause some trouble for users who divide some vector by these quantities, as it will return an absurdly large number instead of `inf` or `nan`. I might need to add a warning to the documentation regarding this.

Reply:
> Since osyris is using a rotation matrix, there's going to be very slight deviations from the analytical formulas

Isn't that what `allclose` is for?

In addition, I still don't see the value of having such a test if we are just copy and pasting the formulas that are in the implementation.
Instead, pick a vector that is e.g. horizontal, or at 45 degrees and then make an easy rotation, say by 90 degrees (or 45 or 60 if you feel adventurous) and then compute the numbers (using `np.pi` in the expected values rather than long float numbers).